### PR TITLE
Add option for Supervisor HTTP gateway authentication token

### PR DESCRIPTION
**Pull request by rarenerd**
### Description

Adds options to the `hab_sup`, `hab_service` and `hab_config` resources to enable authentication on the HTTP gateway.

Fixed some Cookstyle errors along the way.

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

File: libraries/provider_hab_package.rb [hunk 1]
```diff
@@ -114,8 +114,6 @@ def platform_target
             'target=x86_64-windows'
           elsif platform_family?('rhel') && node['platform_version'].to_f < 6.0
             'target=x86_64-linux-kernel2'
-          elsif platform_family?('centos') && node['platform_version'].to_f < 6.0
```
Comment (jonlives) [inline libraries/provider_hab_package.rb]: @sirajrauff why were these lines removed? They seem unrelated to the rest of the changes in this PR.

Comment (rarenerd) [reply libraries/provider_hab_package.rb]: This came up in the latest Cookstyle, 'centos' isn't a platform family.

Comment (jonlives) [reply libraries/provider_hab_package.rb]: Ah that totally makes sense, thanks :D
